### DISCUSSION TITLE
Mirth 4.5.0 DICOM channel, ${message.transformedData} returns NULL

### DISCUSSION THREAD

**Discussion by sglanger** [nextgenhealthcare/connect#6140] [Q&A]
HI All; long time use of an early Mirth 3.x version and this line worked fine
>var msg = new XML(messageObject.getTransformedData().toString());

but in 4.5.0 get a lot of errors. To figure this out I did a file writer and into the message template put ${message.transformedData}  and get nothing

When I replace that with ${message.encodedData} I get a binary dump, which is good, but not what I need. What all changed from 3.2 to this version and how do I get the transformedData back?

**Comment by jonbartels** [comment]
> but in 4.5.0 get a lot of errors. 

What errors do you get?

**Reply by sglanger** [reply]
In 4.0.5 the formerly working line gives;

"[2024-03-21 13:08:02.090]  INFO  (db-connector:138): Exception in mirth:main 
ReferenceError: "messageObject" is not defined."

**Reply by jonbartels** [reply]
Look in `mirth.log` it will often provide more detail in the log file than in the front-end client.

**Reply by sglanger** [reply]
well, it has more but the triggering event is the same

***************************
                                                           DETAILS:	TypeError: Cannot find function substring in object ReferenceError: "messageObject" is not defined..
                                                                               	at a41f3271-b1a0-46ff-88f3-2471691c12be_Database_Writer_1:503 (doScript)
                                                           	at a41f3271-b1a0-46ff-88f3-2471691c12be_Database_Writer_1:774
                                	at org.mozilla.javascript.ScriptRuntime.constructError(ScriptRuntime.java:4280) ~[rhino-1.7.13.jar:1.7.13]
                                                     	at org.mozilla.javascript.ScriptRuntime.constructError(ScriptRuntime.java:4258) ~[rhino-1.7.13.jar:1.7.13]
                                                                     	at org.mozilla.javascript.ScriptRuntime.typeError(ScriptRuntime.java:4291) ~[rhino-1.7.13.jar:1.7.13]
                                                                                	at org.mozilla.javascript.ScriptRuntime.typeError2(ScriptRuntime.java:4310) ~[rhino-1.7.13.jar:1.7.13]
    	at org.mozilla.javascript.ScriptRuntime.notFunctionError(ScriptRuntime.java:4382) ~[rhino-1.7.13.jar:1.7.13]
                       	at org.mozilla.javascript.ScriptRuntime.getPropFunctionAndThisHelper(ScriptRuntime.java:2584) ~[rhino-1.7.13.jar:1.7.13]
******************************************************

Let me ask something a bit different. Has anyone, with the current Mirth, been able to take in a  DICOM image and do a FileWriter out like the below template? Everything works EXCEPT the transformedData

${date.get('yyyy-M-d H.m.s')}
${patientsName}
${patientId}
${accessionNumber}
${sourceApplicationEntityTitle}
${manufacturer}
${manufacturersModelName}
${softwareVersions}
${message.transformedData}

**Comment by tonygermano** [comment]
I'm pretty sure that `messageObject` is from mirth 2.x, not 3.x. It does not exist anymore, and I believe the current equivalent is `connectorMessage`. At any rate, if you set your inbound data type appropriately, there is no need for that kind of manual initialization of `msg`. 

The most likely cause of transformedData being empty is that your transformer was skipped because there were no steps defined and the inbound/outbound settings did not force any transformations.

**Comment by sglanger** [comment]
OK, I migrated the original channel to Mirth 4.5 and imported the 38 step  Source  transformer  along with it. And yet, when I use the above template, the transformed Data dumps nothing. If I replace it with Raw or Encoded, I get a Hex dump. Why?

In other oddness "var loc_result = dbConn.executeUpdate (str, params);" no longer works, it only handles one argument. But "dbConn.executeCachedQuery(str, param);" still does take 2 args

**Reply by pacmano1** [reply]
> In other oddness "var loc_result = dbConn.executeUpdate (str, params);" no longer works, it only handles one argument. But "dbConn.executeCachedQuery(str, param);" still does take 2 args

Not true.

**Comment by yonajj** [comment]
Hello @sglanger could you explain a little more about what are you trying to get? For example if you have a Dicom Listener and you want to write to a file you have to use ${DICOMMESSAGE} instead of ${message.encodedData}
